padlockcloud/server.go

```diff
@@ -477,6 +482,47 @@ func (server *Server) Init() error {
 
 	server.cleanAuthRequests.Start(24 * time.Hour)
 
+	if server.WhitelistConfig.On {
+
+		_, err := os.Stat(server.WhitelistConfig.Path)
+		if os.IsNotExist(err) {
+			return fmt.Errorf("Whitelist file[%s] does not exist:.\n", server.WhitelistConfig.Path)
+		}
+
+		if server.WhitelistConfig.JobUpdateTime.Nanoseconds() <= 0 {
+			return fmt.Errorf("JobUpdateTime must be greater than 0 nanoseconds.\n")
+		}
+
+		server.updateWhitelist = &Job{
+			Action: func() {
+				file, err := os.Open(server.WhitelistConfig.Path)
+				if err != nil {
+					server.Log.Error.Printf("Error opening Whitelist file: %s\n", err.Error())
+					return
+				}
+				defer file.Close()
+				scanner := bufio.NewScanner(file)
+
+				var emails []string
+				for scanner.Scan() {
+					emails = append(emails, scanner.Text())
+				}
+				if err := scanner.Err(); err != nil {
+					server.Log.Error.Printf("Error reading in emails, whitelist not set: %s\n", err.Error())
+					return
+				}
+
+				server.Whitelist.SetEmails(emails)
+				server.Log.Info.Printf("%d Whitelist emails set.\n", len(emails))
+			},
+		}
+
+		//update whitelist at startup
+		server.updateWhitelist.Action()
+
+		server.updateWhitelist.Start(server.WhitelistConfig.JobUpdateTime)
+	}
+
```

Comment:
A better place for this whole chunk would be a function in `whitelist.go`. Something like `func (w *Whitelist) CreateUpdateJob() *Job`